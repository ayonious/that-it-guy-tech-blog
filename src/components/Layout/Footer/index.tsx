import React, { useContext } from "react";
import { VscGithubInverted } from "react-icons/vsc";

import { Context as ThemeContext } from "../../../Context/Theme/ThemeContext";
import {
  IconWrapper,
  LinkCenter,
  NavBar,
  NavCenter,
  NavHeader,
  SpanWrapper,
} from "./styles";

const Footer = () => {
  const {
    state: { theme },
  } = useContext(ThemeContext);

  return (
    <NavBar>
      <NavCenter theme={theme}>
        <NavHeader>
          <SpanWrapper>
            This is an opensource project you can find the source code in
          </SpanWrapper>
          <LinkCenter href="https://github.com/ayonious/that-it-guy-tech-blog">
            <SpanWrapper>github</SpanWrapper>
            <IconWrapper>
              <VscGithubInverted />
            </IconWrapper>
          </LinkCenter>
        </NavHeader>
      </NavCenter>
    </NavBar>
  );
};

export default Footer;
