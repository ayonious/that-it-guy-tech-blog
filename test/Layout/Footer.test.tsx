/**
 * @jest-environment jsdom
 */

import { render } from "@testing-library/react";
import { expect } from "@jest/globals";
import * as React from "react";
import { ThemeProvider as StyledThemeProvider } from "styled-components";

import Footer from "../../src/components/Layout/Footer";
import { Provider as ThemeProvider } from "../../src/Context/Theme/ThemeContext";
import { Provider as TagsProvider } from "../../src/Context/TagFilter/TagFilterContext";
import { darkTheme } from "../../src/Context/Theme/ThemeEnums";

it("Footer renders", () => {
  const tree = render(
    <StyledThemeProvider theme={darkTheme}>
      <ThemeProvider>
        <TagsProvider>
          <Footer />
        </TagsProvider>
      </ThemeProvider>
    </StyledThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});
