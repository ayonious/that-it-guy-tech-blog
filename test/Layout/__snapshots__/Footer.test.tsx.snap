// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c1 {
  margin: 0 auto;
  max-width: 100vw;
  text-align: center;
  background: rgb(25,25,25);
}

.c4 {
  color: darksalmon;
}

.c2 {
  padding: 1rem 1.25rem;
}

.c5 {
  margin-top: 0.2rem;
}

.c3 {
  padding: 0.2rem 0.2rem;
  text-align: center;
}

.c0 {
  padding: 0 0rem;
  margin-top: auto;
}

<body>
    <div>
      <nav
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <span
              class="c3"
            >
              This is an opensource project you can find the source code in
            </span>
            <a
              class="c4"
              href="https://github.com/ayonious/that-it-guy-tech-blog"
            >
              <span
                class="c3"
              >
                github
              </span>
              <span
                class="c5"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M7.976 0A7.977 7.977 0 0 0 0 7.976c0 3.522 2.3 6.507 5.431 7.584.392.049.538-.196.538-.392v-1.37c-2.201.49-2.69-1.076-2.69-1.076-.343-.93-.881-1.175-.881-1.175-.734-.489.048-.489.048-.489.783.049 1.224.832 1.224.832.734 1.223 1.859.88 2.3.685.048-.538.293-.88.489-1.076-1.762-.196-3.621-.881-3.621-3.964 0-.88.293-1.566.832-2.153-.05-.147-.343-.978.098-2.055 0 0 .685-.196 2.201.832.636-.196 1.322-.245 2.007-.245s1.37.098 2.006.245c1.517-1.027 2.202-.832 2.202-.832.44 1.077.146 1.908.097 2.104a3.16 3.16 0 0 1 .832 2.153c0 3.083-1.86 3.719-3.62 3.915.293.244.538.733.538 1.467v2.202c0 .196.146.44.538.392A7.984 7.984 0 0 0 16 7.976C15.951 3.572 12.38 0 7.976 0z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </a>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": .c1 {
  margin: 0 auto;
  max-width: 100vw;
  text-align: center;
  background: rgb(25,25,25);
}

.c4 {
  color: darksalmon;
}

.c2 {
  padding: 1rem 1.25rem;
}

.c5 {
  margin-top: 0.2rem;
}

.c3 {
  padding: 0.2rem 0.2rem;
  text-align: center;
}

.c0 {
  padding: 0 0rem;
  margin-top: auto;
}

<div>
    <nav
      class="c0"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          <span
            class="c3"
          >
            This is an opensource project you can find the source code in
          </span>
          <a
            class="c4"
            href="https://github.com/ayonious/that-it-guy-tech-blog"
          >
            <span
              class="c3"
            >
              github
            </span>
            <span
              class="c5"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M7.976 0A7.977 7.977 0 0 0 0 7.976c0 3.522 2.3 6.507 5.431 7.584.392.049.538-.196.538-.392v-1.37c-2.201.49-2.69-1.076-2.69-1.076-.343-.93-.881-1.175-.881-1.175-.734-.489.048-.489.048-.489.783.049 1.224.832 1.224.832.734 1.223 1.859.88 2.3.685.048-.538.293-.88.489-1.076-1.762-.196-3.621-.881-3.621-3.964 0-.88.293-1.566.832-2.153-.05-.147-.343-.978.098-2.055 0 0 .685-.196 2.201.832.636-.196 1.322-.245 2.007-.245s1.37.098 2.006.245c1.517-1.027 2.202-.832 2.202-.832.44 1.077.146 1.908.097 2.104a3.16 3.16 0 0 1 .832 2.153c0 3.083-1.86 3.719-3.62 3.915.293.244.538.733.538 1.467v2.202c0 .196.146.44.538.392A7.984 7.984 0 0 0 16 7.976C15.951 3.572 12.38 0 7.976 0z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </a>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
